src/Graph.cpp

```diff
@@ -0,0 +1,109 @@
+#include "Graph.h"
+#include <algorithm>
+#include <queue>
+#include <iostream>
+#include <cfloat>
+
+std::vector<Node*> Graph::getNodes() {
+    return nodes_;
+}
+
+unsigned int Graph::getSize() {
+    return nodes_.size();
+}
+
+Node* Graph::getNode(std::string zone) {
+    for (unsigned i = 0; i < nodes_.size(); ++i) {
+        if (nodes_.at(i)->getZone() == zone) {
+            return nodes_.at(i);
+        }
+    }
+    return NULL;
+}
+
+Graph::Graph(std::vector<Node*> nodes) {
+    nodes_ = nodes;
+}
+
+std::vector<std::pair<Node*, double>> Graph::findPathLengths(Node* start, int metric) {
```

Comment:
for dijkstra, would it be a lot cleaner and easier to just make a new class dedicated to it? I'm thinking it would be easy to follow and implement test cases for it.

Reply:
also, i will let you merge, but on the next PR, can you please document within the function so we know what is happening?